include/ttmlir/Target/Utils/MLIRToFlatbuffer.h

```diff
@@ -443,12 +443,9 @@ encodingToFlatbuffer(FlatbufferObjectCache &cache, Attribute attr,
                                   deviceAttr);
   }
 
-  if (isa<ttnn::TTNNLayoutAttr>(attr)) {
-    return ttnnLayoutAttrToFlatbuffer(cache, cast<ttnn::TTNNLayoutAttr>(attr),
-                                      logicalShape, deviceAttr);
-  }
-
-  assert(false && "unsupported layout attr");
```

Comment:
Maybe somehow `assert(false` compiler infers the codepath is unreachable.  This does seem odd.

Reply:
I've figured out what's the diff: in `tt-mlir` even `Release` builds have asserts (due to the `-UNDEBUG` flag), however `Release` builds on `tt-forge-fe` don't have this flag, just `-DNDEBUG`.

I am worried that the `mlir` library depends on these assertions, since they are always on; which seems like a weird design choice for users of the headers. I'll look into this a bit more, but in the meanwhile I would like to get this in to unblock uplift... @nsmithtt sounds good?

Reply:
Sounds good, I agree, we shouldn't be relying on assertions for error handling in release builds.